prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -866,21 +862,91 @@ class Hl7Serializer(
                     }
                 }
             }
-            Element.Type.CODE -> setCodeComponent(terser, value, pathSpec, element.valueSet)
+            Element.Type.CODE -> setCodeComponent(terser, truncatedValue, pathSpec, element.valueSet)
             Element.Type.TELEPHONE -> {
-                if (value.isNotEmpty()) {
+                if (truncatedValue.isNotEmpty()) {
                     val phoneNumberFormatting = hl7Config?.phoneNumberFormatting
                         ?: Hl7Configuration.PhoneNumberFormatting.STANDARD
-                    setTelephoneComponent(terser, value, pathSpec, element, phoneNumberFormatting)
+                    setTelephoneComponent(terser, truncatedValue, pathSpec, element, phoneNumberFormatting)
                 }
             }
             Element.Type.EMAIL -> {
-                if (value.isNotEmpty()) {
-                    setEmailComponent(terser, value, element, hl7Config)
+                if (truncatedValue.isNotEmpty()) {
+                    setEmailComponent(terser, truncatedValue, element, hl7Config)
                 }
             }
-            Element.Type.POSTAL_CODE -> setPostalComponent(terser, value, pathSpec, element)
-            else -> terser.set(pathSpec, value)
+            Element.Type.POSTAL_CODE -> setPostalComponent(terser, truncatedValue, pathSpec, element)
+            else -> terser.set(pathSpec, truncatedValue)
+        }
+    }
+
+    /**
+     * Truncate [value] to maximum length allowed by HL7 specification for a component.
+     * [hl7Field] determines the HL7 component. [hl7Config] enables special truncation rules.
+     */
+    internal fun truncateValue(
```

Comment:
@RickHawesUSDS - will this new truncation logic affect all HL7 fields to truncate to the HL7 spec per the HAPI library?  If so, this might be problematic because [HHS (confluence)](https://confluence.hl7.org/display/OO/Proposed+HHS+ELR+Submission+Guidance+using+HL7+v2+Messages) guidance conflicts with the HL7 specification in some cases.  For example, the HL7 spec has a length limit of 20 for `OBX-17.1`, whereas the HHS guidance is to use the LIVD table Model ID (plus concatenate other fields), which more often than not exceeds 20 characters.  I believe the Reportstream `obx17 ` mapper concatenates `TestKitNameID` and `TestKitNameIDType`, which in most cases will be more than 20 characters.  Example - today we send `BD Veritor System for Rapid Detection of SARS-CoV-2*_EUA` in OBX-17.1 (56 chars), but if we apply the HL7 truncation rules, moving forward we would send `BD Veritor System fo` (20 chars).  I would imagine if we implemented this it would break mappings that STLTs have set up for OBX 17.1.  CC @rhood23699

Reply:
I tend to agree. We should not truncate as much as possible. Look at the current approach which creates a general truncation mechanism for receiver-specific problems.

Reply:
Thanks!  I saw the configurator that allows us to specify the list of HL7 fields that we want to be truncated - that is perfect.